http/check_service.go

```diff
@@ -235,6 +237,28 @@ func (h *CheckHandler) handleGetChecks(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *CheckHandler) handleGetCheckQuery(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	h.Logger.Debug("check query retrieve request", zap.String("r", fmt.Sprint(r)))
+	id, err := decodeGetCheckRequest(ctx, r)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+	chk, err := h.CheckService.FindCheckByID(ctx, id)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+	flux, err := chk.GenerateFlux()
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+	h.Logger.Debug("check query retrieved", zap.String("check query", flux))
+	fmt.Fprintln(w, flux)
```

Comment:
The errors are JSON and the output is text.  This feels unpredictable as an API.  Perhaps respond with JSON?

Reply:
Our "/write" endpoint shows text/plain as response, if any error happens it's still an json. 
Do you think it's worth to do something similar to telegraf response? Based on the "Accept" header we show different responses?

Reply:
/write should not be a text/plain I wouldn't think.

Errors being returned from /write should be in JSON.

An `Accept` header would be the way to handle this if you really think it is important to have plaintext.  However, if you do plaintext you would need an error representation that was also plaintext.